stream-chat-android-client/src/main/java/io/getstream/chat/android/client/call/CoroutineCall.kt

```diff
@@ -0,0 +1,36 @@
+package io.getstream.chat.android.client.call
+
+import io.getstream.chat.android.client.internal.InternalStreamChatApi
+import io.getstream.chat.android.client.utils.Result
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+
+@InternalStreamChatApi
+public class CoroutineCall<T : Any>(
+    private val scope: CoroutineScope,
+    private val runnable: suspend () -> Result<T>
+) : Call<T> {
+
+    private var job: Job? = null
+
+    override fun cancel() {
+        job?.cancel()
+    }
+
+    override fun execute(): Result<T> {
+        return runBlocking { runnable() }
+    }
+
+    override fun enqueue(callback: (Result<T>) -> Unit) {
+        job = scope.launch {
+            val result = runnable()
+            withContext(Dispatchers.Main) {
```

Comment:
As an improvement, we could specify dispatcher for result callback like operator `Flow.flowOn`. But for the first iteration we can omit it, it's not so often case

Reply:
Since clients don't know when they have an instance of `ErrorCall` and when they have some other, cancellable instance, I'd rather just silently not cancel here. Otherwise, cancelling a call (such as `client.queryChannel()`) will sometimes cancel it, and sometimes throw an exception for clients, which would be painful to deal with.